File: py-recitations-solutions/conditionals/pokedex.py
# pokedex.py

# Using a match/case, implement a Pokedex that tells the user the name
# (and description, if you'd like!) of the Gen 1 starters and their
# evolutions.

# Research: https://gamerant.com/pokemon-best-starter-trio-ranked/


# NOTE: Now unfortunately, match/case statements are not a thing until Python 3.10 :/,
# so if you are on a version below, do what you have to do.

GenerationOnePokemon = ["Bulbasaur", "Charmander", "Squirtle"]

# Have user select pokemon starter
select  = int(input("Select a Gen 1 Starter Pokemon : \n0 = Bulbasaur \n1 = Charmander \n2 = Squirtle\n"))

match select:
    case 0:
        print(f"Selected Pokemon : {GenerationOnePokemon[select]}!"
              "\nType: Grass"
              "\nEvolutions: Ivysaur (level 16) -> Venusaur (level 32)")
    case 1:
        print(f"Selected Pokemon : {GenerationOnePokemon[select]}!"
              "\nType: Fire"
              "\nEvolutions: Charmeleon (level 16) -> Charizard (level 36)")
    case 2:
        print(f"Selected Pokemon : {GenerationOnePokemon[select]}!"
              "\nType: Water"
              "\nEvolutions: Wartortle (level 16) -> Blastoise (level 36)")

File: cpp-recitations-solutions/conditionals/pokedex.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
  

  // Using a case/switch, implement a Pokedex that tells the user the name
  // (and description, if you'd like!) of the Gen 1 starters and their
  // evolutions.

  vector<string> Gen1Pokemon = {"Bulbasaur!", "Charmander!", "Squirtle!"};

  int select;

  cout << "Select a Gen 1 Starter Pokemon\n" << "0 = Bulbasaur\n1 = Charmander\n2 = Squirtle\n : ";
  cin >> select;

  string output;

  switch (select)
  {
  case 0:
    output = "Selected Pokemon : " + Gen1Pokemon[0] + "\nType : Grass" + "\nEvolutions: Ivysaur (level 16) -> Venusaur (level 32)";
    break;
  case 1:
    output = "Selected Pokemon : " + Gen1Pokemon[1] + "\nType : Fire" + "\nEvolutions: Charmeleon (level 16) -> Charizard (level 36)";
    break;
  case 2:
    output = "Selected Pokemon : " + Gen1Pokemon[2] + "\nType : Water" + "\nEvolutions: Wartortle (level 16) -> Blastoise (level 36)";
    break;
  default:
    cout << "Seems you've made an erroneous input";
    break;
  }  

  cout << endl <<output;
  return 0;
}